tutorial-day-1/library-api/00_Overview/README.md

```diff
@@ -0,0 +1,69 @@
+
+This tutorial has been divided into different parts, each part demonstrates a functionality/feature. For all the tutorials in this section we will use `example.cpp` file. Contents of this file may change for different sections depending upon the feature being demonstrated. With the exception of tutorial `01` for each tutorial you will find that `example.cpp` does not have timemory library calls added to it, these will be added by the attendees following the instructions in the `README`. The directory `solution` in each tutorial contains a modified version of `example.cpp` that has the timemory calls added to it, you can refer to that as well.
+
+## Getting Started
+In order to include timemory library calls in your code we start off by including the required header:
+```cpp
+#include <timemory/library.h>
+```
+### Initializing Timemory Library
+Initialization is optional. The main benefits of initialization are, by default, timemory will encode the name of the executable into the output folder, e.g. ./foo ... + initialization will generate output in timemory-foo-output/. Finalization is somewhat optional but highly recommended because when data is deleted at the termination of an application can be non-deterministic which can lead to segmentation faults. When MPI or UPC++ is being used, finalization should occur before these libraries are finalized and initialization can happen before or after but initialization afterwards is recommended. Multiple calls to initialization are permitted but multiple calls to finalization is not. Below code segment demonstrates the usage of initialization and finalization calls:
+```cpp
+#include <timemory/library.h>
+int main(int argc, char** argv)
+{
+    timemory_init_library(argc, argv);
+    // ...
+    timemory_finalize_library();
+}
+```
+### Building with Timemory Library
+Simplest way of building with timemory is to use `cmake` to look up the required components. For Timemory library we find the required packages like:
+```cmake
+find_package(timemory REQUIRED COMPONENTS headers compile-options OPTIONAL_COMPONENTS cxx)
+```
+`headers` and `compile-options` include the required headers with standard compilation options for Timemory.
+
+To get the correct library we use the following snippet:
+```cmake
+function(GET_LIBRARY _VAR _LANG)
+        string(TOLOWER ${_LANG} _LOW)
+        set(_PREFIX timemory::timemory-${_LOW})
+        if(TARGET ${_PREFIX}-shared)
+                set(${_VAR} ${_PREFIX}-shared PARENT_SCOPE)
+        elseif(TARGET ${_PREFIX}-static)
+                set(${_VAR} ${_PREFIX}-static PARENT_SCOPE)
+        endif()
+endfunction()
+
+get_library(TIMEMORY_CXX_LIBRARY CXX)
+```
+and then link with the target (in the below example target is name `library_example`):
+```cmake
+target_link_libraries(library_example ${TIMEMORY_CXX_LIBRARY})
+```
```

Comment:
Remove all this and just note that `find_package(timemory ...)` adds all the components to the `timemory::timemory` target or you can do:

```cmake
set(timemory_FIND_COMPONENTS_INTERFACE timemory::library-example)
find_package(timemory REQUIRED COMPONENTS headers compile-options OPTIONAL_COMPONENTS cxx)

# ...
target_link_libraries(library_example timemory::library-example)
```